File: public/examples/java/basic/04_control_flow.java
public class Main {
    public static void main(String[] args) {
        int score = 85;

        // If-Else
        if (score >= 90) {
            System.out.println("Grade: A");
        } else if (score >= 80) {
            System.out.println("Grade: B");
        } else {
            System.out.println("Grade: C");
        }

        // For loop
        System.out.println("Counting:");
        for (int i = 0; i < 3; i++) {
            System.out.println(i);
        }

        // Switch (since Java 14 can use ->)
        String day = "MONDAY";
        switch (day) {
            case "MONDAY": System.out.println("Start of week"); break;
            default: System.out.println("Other day");
        }
    }
}

File: public/examples/csharp/basic/04_control_flow.cs
using System;

// If-Else Statement
int score = 85;
if (score >= 90)
{
    Console.WriteLine("Grade: A");
}
else if (score >= 80)
{
    Console.WriteLine("Grade: B");
}
else
{
    Console.WriteLine("Grade: C");
}

// For Loop
Console.WriteLine("\nCounting 1 to 5:");
for (int i = 1; i <= 5; i++)
{
    Console.WriteLine(i);
}

// While Loop
Console.WriteLine("\nCountdown:");
int count = 3;
while (count > 0)
{
    Console.WriteLine(count);
    count--;
}

// Switch Statement
string day = "Monday";
switch (day)
{
    case "Monday":
        Console.WriteLine("\nStart of the week!");
        break;
    case "Friday":
        Console.WriteLine("\nAlmost weekend!");
        break;
    default:
        Console.WriteLine("\nRegular day");
        break;
}
